docs/modules/SDDs/pages/0021-cluster-catalog-compilation.adoc

```diff
@@ -0,0 +1,170 @@
+= SDD 0021 - Cluster Catalog Compilation
+
+:sdd_author:    Simon Rüegg
+:sdd_owner:
+:sdd_reviewers:
+:sdd_date:      2020-06-10
+:sdd_status:    draft
+include::partial$meta-info-table.adoc[]
+
+[NOTE]
+.Summary
+====
+This document describes how, where and when the xref:0006-commodore-configuration-catalog-compilation.adoc[catalog compilation engine] runs.
+====
+
+
+== Motivation
+
+The compilation of a cluster catalog is a central piece of the configuration management within project Syn.
+It generates the actual configuration for a cluster based on various inputs and configurations.
+It's therefore crucial to have a robust and scalable system in place to run these compilation jobs in a secure and reactive manner.
+
+
+=== Goals
+
+* Define where (in terms of which cluster, namespace, etc.) the jobs are run
+* Define what should cause the recompilation of the catalog for a certain cluster
+* Scale to several hundrets and with future changes to thousands of clusters
+
+
+=== Non-Goals
+
+* The handling of a compiled catalog (for example merge request automation) is out of scope for this design document
+* Solution to automate the component updates
+
+
+== Design Proposal
+
+The design splits the problem into two parts:
+
+. Figuring out which cluster catalogs need to be compiled
```

Comment:
Is the order relevant? If not, make this an unordered list.

Reply:
I added the numbers because it's the order in which they happen.